build_runner/lib/src/asset_graph/serialization.dart

```diff
@@ -121,6 +131,28 @@ class _AssetGraphSerializer {
   }
 }
 
+/// Used to serialize the type of a node using an int.
+enum _NodeTypeId { Source, Synthetic, Generated }
```

Comment:
[nit] I'm not sure 'Id' is useful in these enum names

Reply:
done